flink-table/flink-table-runtime-blink/src/main/java/org/apache/flink/table/runtime/operators/aggregate/GroupAggFunction.java

```diff
@@ -180,7 +179,7 @@ public void processElement(RowData input, Context ctx, Collector<RowData> out) t
 
 			// if this was not the first row and we have to emit retractions
 			if (!firstRow) {
-				if (!stateCleaningEnabled && equaliser.equals(prevAggValue, newAggValue)) {
+				if (stateRetentionTime > 0 && equaliser.equals(prevAggValue, newAggValue)) {
```

Comment:
I think this is the one causes `RetractionITCase#testUniqueProcess` failed. Here should be `stateRetentionTime <= 0` to indicate state cleaning is not enabled.

Reply:
you are write